### PR TITLE
CLC-5218, sign-up eligible CCNs in merged feed; messaging in widget; specs

### PR DESCRIPTION
**Pull request by johncrossman**
https://jira.ets.berkeley.edu/jira/browse/CLC-5215
https://jira.ets.berkeley.edu/jira/browse/CLC-5218

Highlights:
* Bamboo build: https://bamboo.ets.berkeley.edu/bamboo/browse/MYB-CJT55-15
* sign-up eligible CCNs to feed based on what is relevant to the course 
* re-org of webcast/merged.rb for readability
* specs and test data to verify
* new widget message: Webcasts for {{...}} will appear here after classes start

### CODE REVIEW

File: spec/models/webcast/merged_spec.rb [hunk 1]
```diff
@@ -60,6 +64,22 @@
...
+      end
+      before do
+        courses_list = [{
+            :classes=>[{ :sections => [{ :ccn=>'9688', :section_number=>'101', :instruction_format=>'LEC' }] }]
+          }
+        ]
+        expect_any_instance_of(MyAcademics::Teaching).to receive(:courses_list_from_ccns).once.and_return courses_list
+      end
+      it 'pads ccn with zeroes' do
+        expect(feed[:media][2008]['D']['09688']).to be_nil
```
Comment (paulkerschen) [inline spec/models/webcast/merged_spec.rb]: @johncrossman, I might be missing something here, but are you checking for the presence of the '09688' key? This expectation should pass either if the key is present with a nil value, or if the key is absent.

File: src/assets/templates/widgets/webcast.html [hunk 2]
```diff
@@ -5,7 +5,21 @@ <h1 class="cc-visuallyhidden" data-ng-if="isEmbedded">Webcasts</h1>
 
     <div data-ng-bind="proxyErrorMessage"></div>
     <div data-ng-if="!proxyErrorMessage">
-      <div data-ng-if="!videos.length && !audio.length">There are no webcasts available.</div>
+      <div data-ng-if="!videos && !audio">
+        There are no Webcasts scheduled.
+      </div>
+      <div data-ng-if="media">
+        <div data-ng-repeat="(year, term_hash) in media">
```
Comment (christianv) [inline src/assets/templates/widgets/webcast.html]: @johncrossman Can we use camelCase everywhere? e.g. `termHash`?

File: src/assets/templates/widgets/webcast.html [hunk 3]
```diff
@@ -5,7 +5,21 @@ <h1 class="cc-visuallyhidden" data-ng-if="isEmbedded">Webcasts</h1>
...
+          <div data-ng-repeat="(term, ccn_hash) in term_hash">
+            <div data-ng-repeat="(ccn, course) in ccn_hash">
```
Comment (christianv) [inline src/assets/templates/widgets/webcast.html]: @johncrossman interesting that we have to use so many `ngRepeats`, is there a way in the back-end to restructure the format so we can avoid some of those?

File: spec/models/webcast/merged_spec.rb (hunk 1 above)
Comment (johncrossman) [inline spec/models/webcast/merged_spec.rb]: Good catch, @paulkerschen  There was a bug in the associated mock query results. And in my refactoring of test data I made the mistake of expecting nil. Fixed.

File: src/assets/templates/widgets/webcast.html (hunk 2 above)
Comment (johncrossman) [inline src/assets/templates/widgets/webcast.html]: Thanks @christianv  Change made.

File: src/assets/templates/widgets/webcast.html (hunk 3 above)
Comment (johncrossman) [inline src/assets/templates/widgets/webcast.html]: Yes, @christianv. I'm going to create separate Jira and reduce it to strictly a ccnHash.

**Comment by paulkerschen** [comment]
LGTM, and https://jira.ets.berkeley.edu/jira/browse/CLC-5376 is created for the nested repeats.

**Pull request merged**